### PR TITLE
added onClick event to Row widget

### PR DESCRIPTION
**Pull request by sibteali786**
## Describe your changes
- Added onTap event to Row widget 
- Modified hover behavior to pointer
## Issue ticket number and link

Closes #

## Checklist before requesting a review

- [ ] I have performed a self-review of my code
- [ ] I have added tests
- [ ] I have added a changeset `pnpm changeset add`
- [ ] I have added example usage in the kitchen sink app

### CODE REVIEW

File: packages/runtime/src/widgets/Row.tsx [hunk 1]
```diff
@@ -51,8 +61,10 @@ export const Row: React.FC<FlexboxProps> = (props) => {
         flexDirection: "row",
         flexFlow: "unset",
         flexGrow: "unset",
+        cursor: isHovered ? "pointer" : "auto",
```
Comment (evshi) [inline packages/runtime/src/widgets/Row.tsx:64]: You don't need to check for hover state, the browser does that automatically

File: packages/runtime/src/widgets/Row.tsx [hunk 2]
```diff
@@ -26,11 +28,19 @@ export const Row: React.FC<FlexboxProps> = (props) => {
   const renderedChildren = useMemo(() => {
     return children ? EnsembleRuntime.render(children) : null;
   }, [children]);
-
+  const action = useEnsembleAction(props?.onTap);
```
Comment (evshi) [inline packages/runtime/src/widgets/Row.tsx:31]: Destructure this from the props

**Review by evshi — COMMENTED** [review]
Can you also add this to Column since it's part of the spec

File: packages/runtime/src/widgets/Row.tsx (hunk 2 above)
Comment (sibteali786) [reply packages/runtime/src/widgets/Row.tsx:31]: got it @evshi, but i have one question and it is solely for reason of curiosity. If you can answer it please. 
Is there any performance or code quality reason to destructure first and then use vs directly using a prop.

Comment (evshi) [reply packages/runtime/src/widgets/Row.tsx:31]: Yes, in this case if you destructure it first, then it won't be passed into the register bindings so saves extra processing

File: apps/kitchen-sink/src/ensemble/screens/home.yaml [hunk 3]
```diff
@@ -14,6 +14,20 @@ View:
...
+            children:
+              - Text:
+                  text: Testing the Row 
+        - Column:
+            onTap:
+              executeCode:  |
+                console.log('Column Clicked');
+            children:
+              - Text:
+                  text: Testing the Column
```
Comment (evshi) [inline apps/kitchen-sink/src/ensemble/screens/home.yaml:30]: Move this to either actions or widgets screen.

**Review by evshi — APPROVED** [review]
LGTM but looks like you have linter errors

File: packages/runtime/src/widgets/Row.tsx [hunk 4]
```diff
@@ -51,6 +59,7 @@ export const Row: React.FC<FlexboxProps> = (props) => {
         flexDirection: "row",
         flexFlow: "unset",
         flexGrow: "unset",
+        cursor: "pointer",
```
Comment (evshi) [inline packages/runtime/src/widgets/Row.tsx:62]: This should only be pointer if the action callback exists. Same with Column. Can you also update the GridRow with same behavior?

**Comment by anserwaseem** [comment]
@sibteali786 can you also add `visible` property in Row and Column as well?

**Comment by sibteali786** [comment]
> @sibteali786 can you also add `visible` property in Row and Column as well?

jo hukam sarkar

**Review by evshi — APPROVED** [review]

**Pull request merged**